### ISSUE TITLE
Does not build on Solaris-like OSs

### ISSUE THREAD

**Issue by bardo** [nextcloud/notify_push#61]
Hello,

I tried building `notify_push` on SmartOS (a Solaris derivative), but it ultimately failed due to a portability issue in `termion`, which [I reported](https://gitlab.redox-os.org/redox-os/termion/-/issues/189).

Reverse dependency tree:
```
termion v1.5.6
___ source-span v2.2.0
    ___ php-literal-parser v0.2.5
        ___ notify_push v0.1.0
```

**Comment by icewind1991** [comment]
With 0.1.7 you can remove the dependency on `termion` by building with `--no-default-features` (you will loose some coloring in some of the error messages)

**Comment by bardo** [comment]
Feature selection works around the issue, however the build still fails due to https://github.com/timothee-haudebourg/source-span/issues/3.

**Comment by bardo** [comment]
`source-span` has been fixed. Next (and hopefully last) step: `php-literal-parser`:

```
   Compiling php-literal-parser v0.2.8
error[E0599]: no associated item named `Blue` found for unit type `()` in the current scope
   --> /home/admin/.cargo/registry/src/github.com-1ecc6299db9ec823/php-literal-parser-0.2.8/src/error.rs:116:67
    |
116 |                 let mut fmt = Formatter::with_margin_color(Color::Blue);
    |                                                                   ^^^^ associated item not found in `()`

error: aborting due to previous error
```

**Comment by icewind1991** [comment]
fix is releasaed with `php-literal-parser` 0.2.9

**Comment by bardo** [comment]
Just tested and the compilation succeeds with `--no-default-features`, thanks a lot!

Resolving here, in the coming days I will test it and open a new issue in case I encounter problems.
